rskj-core/src/main/java/org/ethereum/rpc/LogFilter.java

```diff
@@ -231,47 +232,75 @@ private static void processSingleBlockByHash(String blockHash, Blockchain blockc
             return;
         }
 
-        long blockNumber = blockByHash.getNumber();
-        processBlocks(blockNumber, blockNumber, filter, blockchain, blocksBloomStore);
+        processBlocks(blockByHash, blockByHash, filter, blockchain, blocksBloomStore);

```

Comment:
If only one block is processed, is there a reason to not use filter.onBlock(..)?

Reply:
Agree, but I will leave it like this (as it was before also) just in case we want to create a bloom even for latest block (regtest mode, whatever).